src/components/PermalinkInput/PermalinkInput.js

```diff
@@ -0,0 +1,151 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { MdContentCopy } from 'react-icons/md';
+import Button from '../Button';
+
+const propTypes = {
+  /**
+   * Value of the permalink
+   */
+  value: PropTypes.string,
+
+  /**
+   * CSS class of the container (input+button).
+   */
+  className: PropTypes.string,
+
+  /**
+   * CSS class of the input field.
+   */
+  classNameInputField: PropTypes.string,
+
+  /**
+   * CSS class of the copy button.
+   */
+  classNameCopyBt: PropTypes.string,
+
+  /**
+   * Content for copy button.
+   */
+  button: PropTypes.any,
+
+  /**
+   * Title for the copy button.
+   */
+  titleCopyBt: PropTypes.string,
+
+  /**
+   * Title for the input field.
+   */
+  titleInputField: PropTypes.string,
+
+  /**
+   * Callback function to get a shortened URL.
+   */
+  getShortenedUrl: PropTypes.func,
+};
+
+const defaultProps = {
+  value: '',
+  className: 'tm-permalink-field',
+  classNameInputField: 'tm-permalink-input',
+  classNameCopyBt: 'tm-permalink-bt',
+  titleCopyBt: '',
+  titleInputField: '',
+  button: <MdContentCopy focusable={false} />,
+  getShortenedUrl: null,
```

Comment:
set a dflt function that returns a promise . With this default value you can get rid of the `if` lin71

Reply:
Done :)